src/Streamly/Coreutils/FileTest.hs

```diff
@@ -415,121 +434,96 @@ groupMatchesEGID =
 -- Comparing with other files
 ------------------------------
 
--- | Returns the result of the comparison function with provided file's
--- modification time as second argument and the that of the file being tested as
--- the first argument
---
-compareModTime :: (POSIXTime -> POSIXTime -> Bool) -> FilePath -> IO FileTest
-compareModTime cmp path = do
+compareTime ::
+       (FileStatus -> POSIXTime)
+    -> (POSIXTime -> POSIXTime -> Bool)
+    -> POSIXTime
+    -> FileTest
+compareTime getFileTime cmp t =
+    FileTest (Predicate (\st -> getFileTime st `cmp` t))
```

Comment:
It should be  (t `cmp`  getFileTime st). The test is giving the opposite result.

Reply:
compareTime is correct. The problem was in compareAge. Fixed it in https://github.com/composewell/streamly-coreutils/pull/71/commits/c8fc400e15510484cabcd18169b678af185c9f07 .